cmplxs/cmplxs_test.go

```diff
@@ -0,0 +1,1340 @@
+		n    int
+		l, u complex128
+		want []complex128
+	}{
+		{
+			n: 5, l: cmplx.Inf(), u: cmplx.Inf(),
+			want: []complex128{cmplx.Inf(), cmplx.Inf(), cmplx.Inf(), cmplx.Inf(), cmplx.Inf()},
+		},
+		{
+			n: 5, l: cmplx.Inf(), u: cmplx.NaN(),
+			want: []complex128{cmplx.Inf(), cmplx.NaN(), cmplx.NaN(), cmplx.NaN(), cmplx.NaN()},
+		},
+		{
+			n: 5, l: cmplx.NaN(), u: cmplx.Inf(),
+			want: []complex128{cmplx.NaN(), cmplx.NaN(), cmplx.NaN(), cmplx.NaN(), cmplx.Inf()},
+		},
+		{
+			n: 5, l: 42, u: cmplx.Inf(),
+			want: []complex128{42, cmplx.Inf(), cmplx.Inf(), cmplx.Inf(), cmplx.Inf()},
+		},
+		{
+			n: 5, l: 42, u: cmplx.NaN(),
+			want: []complex128{42, cmplx.NaN(), cmplx.NaN(), cmplx.NaN(), cmplx.NaN()},
+		},
+		{
+			n: 5, l: cmplx.Inf(), u: 42,
+			want: []complex128{cmplx.Inf(), cmplx.Inf(), cmplx.Inf(), cmplx.Inf(), 42},
+		},
+		{
+			n: 5, l: cmplx.NaN(), u: 42,
+			want: []complex128{cmplx.NaN(), cmplx.NaN(), cmplx.NaN(), cmplx.NaN(), 42},
+		},
+	} {
+		got := Span(make([]complex128, test.n), test.l, test.u)
+		areSlicesSame(t, test.want, got,
+			fmt.Sprintf("Unexpected slice of length %d for %f to %f", test.n, test.l, test.u))
+	}
+}
+
+func TestSub(t *testing.T) {
+	s := []complex128{3 + 2i, 4 + 3i, 1 + 7i, 7 + 1i, 5 - 1i}
+	v := []complex128{1 + 1i, 2 + 4i, 3, 4, 5 - 1i}
+	truth := []complex128{2 + 1i, 2 - 1i, -2 + 7i, 3 + 1i, 0}
+	Sub(s, v)
+	areSlicesEqual(t, truth, s, "Bad subtract")
+	// Test that it panics
+	if !Panics(func() { Sub(make([]complex128, 2), make([]complex128, 3)) }) {
+		t.Errorf("Did not panic with length mismatch")
+	}
+}
+
+func TestSubTo(t *testing.T) {
+	s := []complex128{3 + 2i, 4 + 3i, 1 + 7i, 7 + 1i, 5 - 1i}
+	v := []complex128{1 + 1i, 2 + 4i, 3, 4, 5 - 1i}
+	truth := []complex128{2 + 1i, 2 - 1i, -2 + 7i, 3 + 1i, 0}
+	dst1 := make([]complex128, len(s))
+	dst2 := SubTo(dst1, s, v)
+	areSlicesEqual(t, truth, dst1, "Bad subtract from mutator")
+	areSlicesEqual(t, truth, dst2, "Bad subtract from returned slice")
+	// Test that all mismatch combinations panic
+	if !Panics(func() { SubTo(make([]complex128, 2), make([]complex128, 3), make([]complex128, 3)) }) {
+		t.Errorf("Did not panic with dst different length")
+	}
+	if !Panics(func() { SubTo(make([]complex128, 3), make([]complex128, 2), make([]complex128, 3)) }) {
+		t.Errorf("Did not panic with subtractor different length")
+	}
+	if !Panics(func() { SubTo(make([]complex128, 3), make([]complex128, 3), make([]complex128, 2)) }) {
+		t.Errorf("Did not panic with subtractee different length")
+	}
+}
+
+func TestSum(t *testing.T) {
+	s := []complex128{}
+	val := Sum(s)
+	if val != 0 {
+		t.Errorf("Val not returned as default when slice length is zero")
+	}
+	s = []complex128{3 + 1i, 4 + 2i, 1 + 3i, 7 + 4i, 5 + 5i}
+	val = Sum(s)
+	if val != 20+15i {
+		t.Errorf("Wrong sum returned")
+	}
+}
+
+func randomSlice(l int) []complex128 {
+	s := make([]complex128, l)
+	for i := range s {
+		s[i] = complex(rand.Float64(), rand.Float64())
```

Comment:
Also here, I think we're trying to avoid rand package-level functions.

Reply:
As above.